drgn/commands/_builtin/crash/_net.py

```diff
@@ -0,0 +1,350 @@
+
+from drgn.helpers.linux import net as network
+from drgn.helpers.linux.fs import for_each_file
+from drgn.helpers.linux.list import hlist_for_each_entry
+from drgn.helpers.linux.list import list_for_each_entry
+from drgn.helpers.linux.net import SOCKET_I
+from drgn.helpers.linux.pid import find_task
+from drgn.helpers.common.format import print_table
+from drgn.helpers.linux.sched import task_cpu
+
+from drgn.commands import CommandArgumentError, argument, drgn_argument
+from drgn.commands.crash import CrashDrgnCodeBuilder, crash_command
+
+_NETDEV_HASHBITS = 8
+_NETDEV_HASHENTRIES = 1 << _NETDEV_HASHBITS
+
+sockfamily_dict = {
+    0: "AF_UNSPEC",
+    1: "AF_UNIX",
+    2: "AF_INET",
+    10: "AF_INET6",
+    16: "AF_NETLINK/ROUTE",
+}
+
+socktype_dict = {
+    1: "SOCK_STREAM",
+    2: "SOCK_DGRAM",
+    3: "SOCK_RAW",
+    4: "SOCK_RDM",
+    5: "SOCK_SEQPACKET",
+    6: "SOCK_DCCP",
+    10: "SOCK_PACKET",
+}
+
+hwtype_dict = {
+    0: "ARPHRD_NETROM",
+    1: "ARPHRD_ETHER",
+    2: "ARPHRD_EETHER",
+    3: "ARPHRD_AX25",
+    4: "ARPHRD_PRONET",
+    5: "ARPHRD_CHAOS",
+    6: "ARPHRD_IEEE802",
+    7: "ARPHRD_ARCNET",
+    8: "ARPHRD_APPLETLK",
+    15: "ARPHRD_DLCI",
+    19: "ARPHRD_ATM",
+    23: "ARPHRD_METRICOM",
+    24: "ARPHRD_IEEE1394",
+    27: "ARPHRD_EUI64",
+    32: "ARPHRD_INFINIBAND",
+}
+
+state_dict = {
+    1: "NUD_INCOMPLETE",
+    2: "NUD_REACHABLE",
+    4: "NUD_STALE",
+    8: "NUD_DELAY",
+    16: "NUD_PROBE",
+    32: "NUD_FAILED",
+    64: "NUD_NOARP",
+    128: "NUD_PERMANENT",
+}
+
+
+def netdev_ipv4s(dev: Object) -> List[ipaddress.IPv4Address]:
+    ips = []
+    prog = dev.prog_
+    if dev.ip_ptr:
+        ifa = dev.ip_ptr.ifa_list
+        while ifa:
+            addr_bytes = prog.read(ifa.ifa_address.address_of_(), 4)
+            ips.append(ipaddress.IPv4Address(addr_bytes))
+            ifa = ifa.ifa_next
+    return ips
+
+def netdev_ipv6s(dev: Object) -> List[ipaddress.IPv6Address]:
+    ips = []
+    prog = dev.prog_
+    if dev.ip6_ptr:
+        for addr in list_for_each_entry(
+            "struct inet6_ifaddr",
+            dev.ip6_ptr.addr_list.address_of_(),
+            "if_list",
+        ):
+            addr_bytes = prog.read(addr.addr.in6_u.u6_addr8.address_of_(), 16)
+            ips.append(ipaddress.IPv6Address(addr_bytes))
+    return ips
+
+def for_each_netdev(ns: Object) -> Iterator[Tuple[str, Object]]:
+    try:
+        entry_type = ns.prog_.type("struct netdev_name_node")
+        member = "hlist"
+        entry_is_name_node = True
+    except LookupError:
+        entry_type = ns.prog_.type("struct net_device")
+        member = "name_hlist"
+        entry_is_name_node = False
+
+    for i in range(_NETDEV_HASHENTRIES):
+        head = ns.dev_name_head[i]
+        for dev_node in hlist_for_each_entry(entry_type, head, member):
+            dev = dev_node.dev if entry_is_name_node else dev_node
+            yield (dev_node.name.string_().decode("utf-8"), dev)
```

Comment:
Iterating over the `dev_name_head` hash table is more complicated than necessary (which I realize was a mistake in the `netdev_get_by_name()` helper. I just added simpler `for_each_netdev()` and `netdev_name()` helpers to `drgn.helpers.linux.net` that you can use instead.